Sources/PointFree/Account/Invite.swift

```diff
@@ -122,6 +122,32 @@ let acceptInviteMiddleware: Middleware<StatusLineOpen, ResponseEnded, Tuple2<Tea
         .flatMap(const(conn |> redirect(to: path(to: .account(.index)))))
 }
 
+let addTeammateViaInviteMiddleware: Middleware<
+  StatusLineOpen,
+  ResponseEnded,
+  Tuple2<User?, EmailAddress?>,
+  Data
+  > =
+  filterMap(require1 >>> pure, or: loginAndRedirect)
+    <<< filterMap(require2 >>> pure, or: invalidSubscriptionErrorMiddleware)
+    <<< requireStripeSubscription
+    <<< requireActiveSubscription
+    <| { (conn: Conn<StatusLineOpen, Tuple3<Stripe.Subscription, User, EmailAddress>>) in
+
+      let (stripeSubscription, inviter, email) = lower(conn.data)
+      let newPricing = Pricing(
+        billing: stripeSubscription.plan.interval == .month ? .monthly : .yearly,
+        quantity: stripeSubscription.quantity + 1
+      )
+
+      return conn
+        .map(const((stripeSubscription, newPricing)))
+        |> changeSubscription(
+          error: subscriptionModificationErrorMiddleware,
+          success: map(const(email .*. inviter .*. unit)) >>> sendInviteMiddleware
```

Comment:
here we get to run the shared middleware `changeSubscription` and on success we then run the `sendInviteMiddleware` middleware.